src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -99,6 +99,7 @@ public ValidatingOutputFormat getOutputFormatForRun(BatchSinkContext context) th
   public void prepareRun(BatchSinkContext context) throws Exception {
     super.prepareRun(context);
     String cmekKey = context.getArguments().get(GCPUtils.CMEK_KEY);
+    cmekKey = config.getCmekKey(cmekKey);
```

Comment:
this logic will need to be done in multiple plugins and should be pulled into its own class instead of repeated in every plugin. GCPUtils.CMEK_KEY should be private to that class.

For example, may make sense to put this in the base GCPConfig class.

Reply:
You are mentioning about the getCmekKey() method, right?
If we put this in the GCPConfig class then cmek key will be included as a config in all sink plugins extending it.
Currently, all sink plugins don't support cmek.